### PR TITLE
Hathi online

### PR DESCRIPTION
**Pull request by cdmo**
#204 

- [x] Need to look at how the csv constant is populated, maybe use config gem?
- [x] Need to run an extract and see how this performs, I am right now

## Additional changes

* trims whitespace off of `001` for `id` field
* updates "soft depredations" in traject
* updates `custom_spec.rb` to fix some problems in the organization of the specs
* loosens requirements on and updates Traject gem
* added /spec filter for simplecov
* ran rubocop auto fix after deleting part of the rubocop todo

### CODE REVIEW

File: config/indexer_settings_test.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+
+processing_thread_pool: 5
+hathi_overlap_path: spec/fixtures/hathitrust/
```
Comment (banukutlu) [inline config/indexer_settings_test.yml:4]: 👌

File: lib/traject/psulib_config.rb [hunk 2]
```diff
@@ -25,31 +26,36 @@
...
 settings do
-  provide 'solr.url', SOLR_URL
-  provide 'log.batch_size', indexer_settings['log_batch_size']
-  provide 'solr.version', indexer_settings['solr_version']
-  provide 'log.file', indexer_settings['log_file']
-  provide 'log.error_file', indexer_settings['log_error_file']
-  provide 'solr_writer.commit_on_close', indexer_settings['solr_writer_commit_on_close']
-  provide 'reader_class_name', indexer_settings['reader_class_name']
-  provide 'commit_timeout', indexer_settings['commit_timeout']
+  provide 'solr.url', ENV['SOLR_URL'] || 'http://localhost:8983/solr/psul_blacklight'
```
Comment (banukutlu) [inline lib/traject/psulib_config.rb:32]: 👌

**Comment by cdmo** [comment]
btw I did test this on local and it didn't have any noticeable impact on records per second. just under a 1,000 on my mac

File: spec/lib/traject/macros/custom_spec.rb [hunk 3]
```diff
@@ -11,125 +11,127 @@
...
   end
 
-  describe 'A record where indicator 2 is 0 and magic word is not in one of the label subfields' do
-    let(:url_856_1) do
-      { '856' => { 'ind1' => '0', 'ind2' => '0', 'subfields' => [{ 'u' => 'https://scholarsphere.psu.edu/files/02870v8'\
-                                                                          '5d' },
-                                                                 { 'z' => 'This text is irrelevant' }] } }
-    end
-    let(:result_1) { @indexer.map_record(MARC::Record.new_from_hash('fields' => [url_856_1], 'leader' => leader)) }
+  describe '#extract_link_date' do
```
Comment (banukutlu) [inline spec/lib/traject/macros/custom_spec.rb:14]:
```suggestion
  describe '#extract_link_data' do
```

**Pull request merged**